### PR TITLE
chore: add jest and ci github actions

### PR DESCRIPTION
**Pull request by dinarabs**
- added jest with samples to test workflow pass/fail
- added yml file for test, which should be merged with the one from Oguz to build -> test -> deploy

### CODE REVIEW

**Comment by itzMaffi** [comment]
Like with Oz, are github workflows supposed to be in the root folder or not?

File: Agenda-client/package.json [hunk 1]
```diff
@@ -7,13 +7,18 @@
     "dev": "vite",
     "build": "tsc && vite build",
     "lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
-    "preview": "vite preview"
+    "preview": "vite preview",
+    "test": "jest"
   },
   "dependencies": {
+    "jest-environment-jsdom": "^29.7.0",
```
Comment (itzMaffi) [inline Agenda-client/package.json:14]: Is this needed as a dep? Shouldn't it be a dev dep?

**Review by itzMaffi — COMMENTED** [review]
Left a comment

File: Agenda-client/package.json (hunk 1 above)
Comment (dinarabs) [reply Agenda-client/package.json]: fixed

**Comment by dinarabs** [comment]
@itzMaffi thanks for review! fixed mentioned places, pls review again

**Review by dayt47 — APPROVED** [review]

**Pull request merged**